src/workerd/api/memory-cache.h

```diff
@@ -428,30 +428,16 @@ class MemoryCache: public jsg::Object {
   SharedMemoryCache::Use cacheUse;
 };
 
-// Data structure that maps unique cache identifiers to cache instances.
-// This allows separate isolates to access the same in-memory caches.
-class MemoryCacheMap {
+// The MemoryCacheProvider provides the internal implementation of the MemoryCache mechanism.
+// It is responsible for owning the SharedMemoryCache instances and providing them to the
+// bindings as needed. The default implementation (created and returned by createDefault())
+// uses a simple in-memory map to store the SharedMemoryCache instances.
+class MemoryCacheProvider {
 public:
-  MemoryCacheMap(
-      kj::Maybe<SharedMemoryCache::AdditionalResizeMemoryLimitHandler>
-          additionalResizeMemoryLimitHandler = kj::none)
-      : additionalResizeMemoryLimitHandler(kj::mv(additionalResizeMemoryLimitHandler)) {}
-  KJ_DISALLOW_COPY_AND_MOVE(MemoryCacheMap);
+  virtual SharedMemoryCache& getInstance(kj::StringPtr cacheId) const = 0;
```

Comment:
I know this isn't new, but I'm surprised to see this return a reference. It looks like memory caches are never wholly evicted today? That's going to be a big blocker to actually rolling out this feature. This really ought to return an atomic-refcounted object, and when the last reference is dropped (the last `Worker` using it is destroyed) then the cache should be evicted as well.

Reply:
Yep, I noted that also. Quite happy to make that change.